Flake in `TestTxPool_BroadcastQueue`

See https://github.com/celestiaorg/celestia-core/actions/runs/8210249540/job/22457347611?pr=1260
```
--- FAIL: TestTxPool_BroadcastQueue (0.00s)
    pool_test.go:755: 
        	Error Trace:	/home/runner/work/celestia-core/celestia-core/mempool/cat/pool_test.go:755
        	            				/opt/hostedtoolcache/go/1.22.1/x64/src/runtime/asm_amd64.s:1695
        	Error:      	Not equal: 
        	            	expected: types.Tx{0x73, 0x65, 0x6e, 0x64, 0x65, 0x72, 0x2d, 0x30, 0x30, 0x30, 0x2d, 0x30, 0x3d, 0x33, 0x31, 0x3d, 0x31}
        	            	actual  : types.Tx{0x73, 0x65, 0x6e, 0x64, 0x65, 0x72, 0x2d, 0x30, 0x30, 0x30, 0x2d, 0x30, 0x3d, 0x33, 0x30, 0x3d, 0x31}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,3 +1,3 @@
        	            	 (types.Tx) (len=17) {
        	            	- 00000000  73 65 6e 64 65 72 2d 30  30 30 2d 30 3d 33 31 3d  |sender-000-0=31=|
        	            	+ 00000000  73 65 6e 64 65 72 2d 30  30 30 2d 30 3d 33 30 3d  |sender-000-0=30=|
        	            	  00000010  31                                                |1|
        	Test:       	TestTxPool_BroadcastQueue
```

--- a/mempool/cat/pool_test.go
+++ b/mempool/cat/pool_test.go
@@ -745,6 +745,12 @@ func TestTxPool_BroadcastQueue(t *testing.T) {
 
 	wg := sync.WaitGroup{}
 	wg.Add(1)
+
+	for i := 0; i < txs; i++ {
+		tx := newDefaultTx(fmt.Sprintf("%d", i))
+		require.NoError(t, txmp.CheckTx(tx, nil, mempool.TxInfo{SenderID: 0}))
+	}
+
 	go func() {
 		defer wg.Done()
 		for i := 0; i < txs; i++ {
@@ -754,14 +760,8 @@ func TestTxPool_BroadcastQueue(t *testing.T) {
 			case wtx := <-txmp.next():
 				require.Equal(t, wtx.tx, newDefaultTx(fmt.Sprintf("%d", i)))
 			}
-			time.Sleep(10 * time.Millisecond)
 		}
 	}()
-
-	for i := 0; i < txs; i++ {
-		tx := newDefaultTx(fmt.Sprintf("%d", i))
-		require.NoError(t, txmp.CheckTx(tx, nil, mempool.TxInfo{SenderID: 0}))
-	}
 
 	wg.Wait()
 }

